src/GitHub/Data/Webhooks/Events.hs

```diff
@@ -0,0 +1,1269 @@
+data PushEvent = PushEvent
+    { evPushRef                 :: !Text
+    , evPushHeadSha             :: !(Maybe Text)
+    , evPushBeforeSha           :: !(Maybe Text)
+    , evPushCreated             :: !Bool
+    , evPushDeleted             :: !Bool
+    , evPushForced              :: !Bool
+    , evPushBaseRef             :: !(Maybe Text)
+    , evPushCompareUrl          :: !URL
+    , evPushCommits             :: !(Maybe (Vector HookCommit))
+    , evPushHeadCommit          :: !(Maybe HookCommit)
+    , evPushRepository          :: !HookRepository
+    , evPushOrganization        :: !(Maybe HookOrganization)
+    , evPushSender              :: !HookUser
+    }
+    deriving (Eq, Show, Typeable, Data, Generic)
+
+instance WebHookEvent PushEvent where senderOfEvent = evPushSender
+instance NFData PushEvent where rnf = genericRnf
+
+
+type ReleaseEventAction = Text
+
+releasePublished :: ReleaseEventAction
+releasePublished = pack "published"
+
+-- | Triggered when a release is published.
+-- See <https://developer.github.com/v3/activity/events/types/#releaseevent>.
+data ReleaseEvent = ReleaseEvent
+    { evReleaseEventAction      :: !ReleaseEventAction      -- ^ Currently only releasePublished.
+    , evReleaseEventPayload     :: !HookRelease
+    , evReleaseEventRepo        :: !HookRepository
+    , evReleaseEventSender      :: !HookUser
+    }
+    deriving (Eq, Show, Typeable, Data, Generic)
+
+instance WebHookEvent ReleaseEvent where senderOfEvent = evReleaseEventSender
+instance NFData ReleaseEvent where rnf = genericRnf
+
+
+type RepositoryEventAction = Text
+
+repositoryCreated :: RepositoryEventAction
+repositoryCreated = "created"
+
+repositoryDeleted :: RepositoryEventAction
+repositoryDeleted = "deleted"
+
+repositoryArchived :: RepositoryEventAction
+repositoryArchived = "archived"
+
+repositoryUnarchived :: RepositoryEventAction
+repositoryUnarchived = "unarchived"
+
+repositoryPublicized :: RepositoryEventAction
+repositoryPublicized = "publicized"
+
+repositoryPrivatized :: RepositoryEventAction
+repositoryPrivatized = "privatized"
+
+-- | Triggered when a repository is created, archived, unarchived, made public, or made private.
+-- Organization hooks are also triggered when a repository is deleted.
+-- Events of this type are not visible in timelines. These events are only used to trigger hooks.
+-- See <https://developer.github.com/v3/activity/events/types/#repositoryevent>.
+data RepositoryEvent = RepositoryEvent
+    { evRepositoryAction        :: !RepositoryEventAction
+    , evRepositoryTarget        :: !HookRepository
+    , evRepositoryOrg           :: !(Maybe HookOrganization)
+    , evRepositorySender        :: !HookUser
+    }
+    deriving (Eq, Show, Typeable, Data, Generic)
+
+instance WebHookEvent RepositoryEvent where senderOfEvent = evRepositorySender
+instance NFData RepositoryEvent where rnf = genericRnf
+
+
+type StatusEventState = Text
```

Comment:
and other, can these be `newtype`, or better ADT sums, as there is a finite amount of states.

Reply:
... You could have `StatusEventStateUnknown Text` as catch all to be resilient about unknown stuff, but otherwise I'd prefer not to deal with untyped `Text`.